test/optimizer/plugins/ResolveValueSetComponentRuleURLsOptimizer.test.ts

```diff
@@ -36,49 +33,73 @@ describe('optimizer', () => {
       expect(optimizer.runAfter).toBeUndefined();
     });
 
-    it('should replace filter rule system url with the name of a local CodeSystem', () => {
+    it('should replace filter rule system url with the name of a local CodeSystem', async () => {
       const valueset = new ExportableValueSet('MyValueSet');
       const rule = new ExportableValueSetFilterComponentRule(true);
       rule.from = { system: 'http://example.org/tests/CodeSystem/simple.codesystem' };
       valueset.rules.push(rule);
       const myPackage = new Package();
       myPackage.add(valueset);
+      await restockLake(
+        lake,
+        path.join(__dirname, 'fixtures', 'simple-codesystem.json'),
+        path.join(__dirname, 'fixtures', 'unsupported-codesystem.json'),
+        path.join(__dirname, 'fixtures', 'simple-valueset.json'),
+        path.join(__dirname, 'fixtures', 'unsupported-valueset.json')
+      );
+      fisher = new MasterFisher(lake, defs);
```

Comment:
I expect that I am missing something, but... why do we have to constantly restock the lake in every test? Why don't we stock it in `beforeAll` like we used to? I don't see any modifications to the resources in the lake -- but maybe I'm missing it.

If this does indeed need to happen in every test, I think you should extract it into a function like `getRestockedFisher` (or something like that) since it looks like this logic is repeated a lot across the tests.

Reply:
The set of resources is slightly different in some of the tests, which is why I pulled it out of `beforeAll`. I'll make a helper function like you suggested.